### PR TITLE
Hotfix/1690

### PR DESCRIPTION
**Pull request by mehul0810**
## Description
This PR is for #1690.

This issue was not working due to sitemap cache of Yoast SEO. So, i've implemented a change in the PR which will clear sitemap cache on change of admin settings of Give Core as mentioned in https://github.com/Yoast/wordpress-seo/issues/2463  

## How Has This Been Tested?
I've tested this issue with new and fresh WP instance with active/inactive Yoast SEO plugin and All in one SEO plugin.

## Types of changes
Bug fix (non-breaking change which fixes an issue)

## Checklist:
- [x] My code is tested.
- [x] My code follows the WordPress code style.
- [x] My code has proper inline documentation.

### CODE REVIEW

File: includes/actions.php [hunk 1]
```diff
@@ -245,3 +245,19 @@ function give_set_donation_levels_max_min_amount( $form_id ) {
...
+ * This function will clear seo sitemap cache on update of settings
+ *
+ * @since 1.8.9
+ *
+ * @return void
+ */
+function give_clear_seo_sitemap_cache_on_settings_change() {
+	// If Yoast SEO plugin exists, then update seo sitemap cache.
+	if( is_plugin_active( 'wordpress-seo/wp-seo.php' ) ) {
+		WPSEO_Sitemaps_Cache::register_clear_on_option_update( 'give_settings', '' );
```
Comment (ravinderk) [inline includes/actions.php]: @mehul0810 Can we make it more specific means do not delete sitemap cache on every `give_settings` update.  Clear cache only when `Setting --> Display Options --> Post Types` setting changes.

File: give.php [hunk 2]
```diff
@@ -356,6 +356,8 @@ private function setup_constants() {
 		private function includes() {
 			global $give_options;
 
+			require_once ABSPATH . 'wp-admin/includes/plugin.php';
```
Comment (ravinderk) [inline give.php]: @mehul0810 Why do we need this file?

Comment (mehul0810) [inline give.php]: @ravinderk This file is required for `is_plugin_active()` to work. It is not working without this file.

File: includes/actions.php (hunk 1 above)
Comment (mehul0810) [inline includes/actions.php]: @ravinderk As we have all the data of admin settings saved under a single option, I've used it `WPSEO_Sitemaps_Cache::register_clear_on_option_update( 'give_settings', '' );`. But, I'll check if there is any way out to manage this in a way you suggested and let you know.

Comment (ravinderk) [inline includes/actions.php]: @mehul0810  use this hook https://github.com/WordImpress/Give/blob/master/includes/admin/class-admin-settings.php#L116

File: give.php (hunk 2 above)
Comment (ravinderk) [inline give.php]: @mehul0810 can we wrap file require:
```
if( ! funciton_exists('is_plugin_active') ){
    require_once ABSPATH . 'wp-admin/includes/plugin.php';
}
```

It is more safe to load file this way.

File: includes/actions.php (hunk 1 above)
Comment (mehul0810) [inline includes/actions.php]: @ravinderk I've made changes as per the hook suggested.

File: give.php (hunk 2 above)
Comment (mehul0810) [inline give.php]: @ravinderk Yes, we can do this as well.. I'll implemented this change. Also, i've added support for Yoast SEO Premium.

File: includes/actions.php [hunk 3]
```diff
@@ -245,3 +245,23 @@ function give_set_donation_levels_max_min_amount( $form_id ) {
...
 add_action( 'give_pre_process_give_forms_meta', 'give_set_donation_levels_max_min_amount', 30 );
+
+/**
+ * This function will clear seo sitemap cache on update of settings
+ *
+ * @since 1.8.9
+ *
+ * @return void
+ */
+function give_clear_seo_sitemap_cache_on_settings_change() {
```
Comment (ravinderk) [inline includes/actions.php]: @mehul0810 I have following suggestions:
1. Clear sitemap cache only when the only form related settings change.
    for example, single form pageview setting edit or form archive setting edit
2. Add `is_admin` check and move this code to `admin/admin-actions.php`

**Pull request merged**